pytorch_lightning/core/memory.py

```diff
@@ -225,11 +227,13 @@ def param_nums(self) -> List[int]:
 
     @property
     def total_parameters(self) -> int:
-        return sum(p.numel() for p in self._model.parameters())
+        return sum(p.numel() if not _is_lazy_weight_tensor(p) else 0 for p in self._model.parameters())
```

Comment:
Would be good to add a warning that lazy_weight have been detected and won't be counted for reporting the total number of parameters count.

Reply:
Here is how you can do such thing: https://github.com/PyTorchLightning/pytorch-lightning/blob/ca6c87ffbe23957df0f317a2f5b322c7186b505e/pytorch_lightning/plugins/precision/deepspeed_precision.py#L66

Create a warningCache.
Use the warning_cache to warn the user once about the total number of parameters reported not being fully accurate due to lazy module.

Reply:
Hey @tchaton. Thanks for your input. Sorry for my unawareness but where should that `warning.cache` statement go? I am assuming in the summary class. Is that correct?

Reply:
I have added the warning in the `_is_lazy_weight_tensor` function